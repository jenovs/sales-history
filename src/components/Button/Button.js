import React from 'react';
import PropTypes from 'prop-types';

import Btn from './styled';

const propTypes = {
  inactive: PropTypes.bool,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

const defaultProps = {
  inactive: false,
};

const Button = ({ inactive, children, onClick, ...props }) => (
  <Btn type="button" onClick={onClick} inactive={inactive} {...props}>
    {children}
  </Btn>
);

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
