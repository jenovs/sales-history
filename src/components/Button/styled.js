import styled from '@emotion/styled';
import { darken } from 'polished';

export default styled.button`
  background-color: ${p => (p.inactive ? 'white' : 'cornflowerblue')};
  border: 2px solid cornflowerblue;
  border-radius: 2px;
  color: ${p => (p.inactive ? darken(0.2, 'cornflowerblue') : 'white')};
  cursor: inherit;
  font-size: 1.2em;
  margin: 0.2rem;
  padding: 0.1rem 0.5rem;
  transition: all 0.15s;
  width: ${p => (p.width ? `${p.width}px` : 'auto')};
`;

// background-color: ${p =>
//   !p.inactive ? darken(0.1, 'cornflowerblue') : 'white'};
