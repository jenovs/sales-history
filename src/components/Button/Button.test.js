import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';
import 'jest-dom/extend-expect';

import Button from '.';

describe('Button', () => {
  afterEach(cleanup);

  const mockClick = jest.fn();

  const props = {
    onClick: () => {},
  };

  it('should render a button with a label', () => {
    const { getByText } = render(<Button {...props}>Foo</Button>);

    expect(getByText('Foo')).toBeVisible();
  });

  it('should register a click', () => {
    const { getByText } = render(
      <Button {...props} onClick={mockClick}>
        Foo
      </Button>
    );

    fireEvent.click(getByText('Foo'));
    expect(mockClick).toHaveBeenCalled();
  });

  it('should render a dark button', () => {
    const tree = renderer.create(<Button {...props}>Foo</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render a light button', () => {
    const tree = renderer
      .create(
        <Button {...props} inactive>
          Foo
        </Button>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
