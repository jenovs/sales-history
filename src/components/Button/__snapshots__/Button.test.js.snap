// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a dark button 1`] = `
<button
  className="css-1lnopfn"
  onClick={[Function]}
  type="button"
>
  Foo
</button>
`;

exports[`Button should render a light button 1`] = `
<button
  className="css-1jabc46"
  onClick={[Function]}
  type="button"
>
  Foo
</button>
`;
